apps/veil/src/pages/tournament/server/specific-delegator-rewards.ts

```diff
@@ -0,0 +1,98 @@
+// The code in this file will have direct access to pindexer.
+'use server';
+
+import { pindexerDb } from '@/shared/database/client';
+import { deserialize, serialize, Serialized } from '@/shared/utils/serializer';
+import { Address } from '@penumbra-zone/protobuf/penumbra/core/keys/v1/keys_pb';
+import { sql } from 'kysely';
+import { DelegatorReward } from '../model/rewards';
+import { AssetId, Value } from '@penumbra-zone/protobuf/penumbra/core/asset/v1/asset_pb';
+import { STAKING_TOKEN_ASSET_ID } from '@/shared/api/fetch-registry';
+import { pnum } from '@penumbra-zone/types/pnum';
+
+export type SortKey = 'epoch' | 'reward';
+
+export type SortDirection = 'desc' | 'asc';
+
+export interface SpecificDelegatorRewardsRequest {
+  address: Address;
+  limit?: number;
+  page?: number;
+  sortKey?: SortKey;
+  sortDirection?: SortDirection;
+}
+
+export interface SpecificDelegatorRewardsResponse {
+  rewards: DelegatorReward[];
+  total: Value;
+  count: number;
+}
+
+async function fetchRewards(
+  address: Address,
+  limit?: number,
+  page?: number,
+  sortKey?: SortKey,
+  sortDirection?: SortDirection,
+): Promise<DelegatorReward[]> {
+  const result = await pindexerDb
+    .selectFrom('lqt.delegator_history')
+    .select([
+      'epoch',
+      'asset_id',
+      'reward',
+      // The power of this delegator in that epoch, divided by the power over the entire epoch.
+      // TODO: upstream this into pindexer.
+      eb =>
+        sql<number>`power / ${eb.fn.sum('power').over(ob => ob.partitionBy('epoch'))}`.as('share'),
+    ])
+    .where('address', '=', Buffer.from(address.inner))
+    .orderBy(sortKey ?? 'epoch', sortDirection ?? 'desc')
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- Kysely limitation
+    .$if(limit !== undefined, qb => qb.limit(limit!))
+    // eslint-disable-next-line @typescript-eslint/no-non-null-assertion -- Kyseley limitation
+    .$if(page !== undefined && limit !== undefined, qb => qb.offset(page! * limit!))
```

Comment:
question: why not giving default values to limit & page?

Reply:
For the page, you could make it 0. For the limit, there's no obvious value. Infinity doesn't work well, because then you'd need to do 0 * Infinity, which is NaN